src/store/sagas/AuthSagas.js

```diff
@@ -23,6 +23,11 @@ export function* userRegister({ payload }) {
     yield put(push('/login'));
     yield put(go());
   } catch (error) {
-    yield put(registerError(true));
+    yield put(registerError({flag: true, message: extractErrorMessage(error)}));
   }
 }
+
+function extractErrorMessage(error) {
+  const errorData = error.response.data;
+  return errorData[Object.keys(errorData)[0]];
```

Comment:
ovo mi je malo magicno, jer zamisli da se igrom slucaja zamjeni mjesto pozicije tog kljuca, sta bi se desilo onda ? Ako vec iz odredjenog kljuca cupas nesto onda eksplicitno njega i navedi ovdje

Reply:
Meni BE vraca kljuceve spram toga u kom polju se desila greska. Npr. ako je greska vezana za email, kljuc ce biti email, tako i za ostala polja. Zbog toga nisam zeleo da se vezem za naziv kljuca vec da univerzalno uzmem prvu gresku na koju naidje (posto mi moze vratiti vise kljuceva).

Reply:
Svakako se slazem da je malo vise magicno i da nije bas najbolji pristup.
Mogao bih onda da proveravam postojanje kljuceva email, password i name posto su mi to polja na kojima moze doci do greske

Reply:
aha aha, jasno mi je sad, mozemo i ostaviti ovako onda